### PR TITLE
Add start script and local db config

### PR DESCRIPTION
**Pull request by bryophyta**
## What does this change?

- Add a config function for connecting to the local postgres db
- Add a start script (modelled after [the Grid's `start.sh`](https://github.com/guardian/grid/blob/main/dev/script/start.sh) which:
  - Checks for necessary dependencies
  - By default will spin up the Docker containers, run the `ingestion-lambda` in 'dev' mode, and start the Play app.
  - If passed the `--use-CODE` flag, it will establish a tunnel to the CODE db, if there isn't one already, and start the Play app.

## Accessibility

-   [ ] [Tested with screen reader](https://github.com/guardian/accessibility/blob/main/people-and-technology/03-visual.md#screen-reader)
-   [ ] [Navigable with keyboard](https://github.com/guardian/accessibility/blob/main/people-and-technology/02-physical.md#keyboard)
-   [ ] [Colour contrast passed](https://github.com/guardian/accessibility/blob/main/people-and-technology/03-visual.md#contrast)
-   [ ] [The change doesn't use only colour to convey meaning](https://github.com/guardian/accessibility/blob/main/people-and-technology/03-visual.md#use-of-colour)

### CODE REVIEW

File: newswires/app/AppComponents.scala [hunk 1]
```diff
@@ -46,8 +46,11 @@ class AppComponents(context: Context)
     .build()
 
   if (context.environment.mode == Mode.Dev) {
-    // TODO run against a local DB?
-    Database.configureRemoteDevDb(ssmClient)
+    if (sys.env.contains("USE_CODE_DB")) {
```
Comment (bryophyta) [inline newswires/app/AppComponents.scala:49]: not super happy about using an environment variable here

File: scripts/start [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+}
+
+tunnelToAwsDb() {
+    if (docker stats --no-stream &> /dev/null); then
+      docker compose down
+    fi
+    if [[ -n $EXISTING_TUNNELS ]]; then
+      echo "RE-USING EXISTING TUNNEL TO CODE POSTGRES (on port ${POSTGRES_PORT})"
+    else
+      # todo: Grid script uses the following options, do we need any of these?
```
Comment (bryophyta) [inline scripts/start:82]: I need to look into this still

File: scripts/start [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+    hasCredentials
+    checkRequirements
+    checkNodeVersion
+    if [ $USE_CODE == true ]; then
+        echo "Using CODE. Starting play app with tunnel to AWS DB."
+        tunnelToAwsDb && USE_CODE_DB=true startPlayApp
+    else
+      echo "Using the local stack. Starting Docker, Ingestion Lambda and Play App."
+      # run all three, and terminate together when the user presses `ctrl+c`: https://stackoverflow.com/a/52033580
+      (trap 'kill 0' SIGINT; startDockerContainers && startIngestionLambda & startPlayApp & wait)
```
Comment (bryophyta) [inline scripts/start:154]: this seems to broadly work as expected, but I've noticed a couple of bugs with my shell when subsequently doing interactive stuff in the same shell session, so I think this needs a bit more investigation..

File: newswires/app/AppComponents.scala (hunk 1 above)
Comment (twrichards) [reply newswires/app/AppComponents.scala:49]: we discussed in-person and ideally we might instead have an environment variable more specific to connect differences, such as `USE_IAM_AUTH` but let's save for a future PR

File: scripts/start (hunk 3 above)
Comment (twrichards) [reply scripts/start:147]: I've had trouble with backgrounding via `&` before, let's see how painful it gets when using day-to-day and improve in a future PR if needs be

**Review by twrichards — APPROVED** [review]

**Comment by twrichards** [comment]
Awesome work @bryophyta this will make things much easier while we're developing 🙌

**Pull request merged**